lib/core/src/animation/animator/continuous.rs

```diff
@@ -84,24 +86,28 @@ impl ContinuousAnimatorData {
 // === ContinuousAnimator ===
 // ==========================
 
-/// This structure runs an animation with continuous time as its input.
+/// This structure runs an animation with continuous time in milliseconds.
 pub struct ContinuousAnimator {
     data : Rc<ContinuousAnimatorData>
 }
 
 impl ContinuousAnimator {
-    pub fn new<F:AnimationCallback>(event_loop:&mut AnimationFrameLoop, f:F) -> Self {
+    /// Creates `ContinuousAnimator` with an `AnimationCallback`.
+    pub fn new<F:AnimationCallback>(event_loop:&mut EventLoop, f:F) -> Self {
         let data            = ContinuousAnimatorData::new(f);
-        let data_clone      = data.clone();
+        let weak_data       = Rc::downgrade(&data);
         let callback_guard  = event_loop.add_callback(move |current_time| {
-            let absolute_start_ms = data_clone.absolute_start_ms();
-            let absolute_start_ms = absolute_start_ms.unwrap_or_else(|| {
-                data_clone.set_absolute_start_ms(Some(current_time));
-                current_time
-            });
-            let relative_start_ms = data_clone.relative_start_ms();
-            let relative_time_ms  = current_time - absolute_start_ms + relative_start_ms;
-            data_clone.on_animation_frame(relative_time_ms);
+            if let Some(data) = weak_data.upgrade() {
+                let absolute_start_ms = data.absolute_start_ms();
+                let absolute_start_ms = absolute_start_ms.unwrap_or_else(|| {
+                    data.set_absolute_start_ms(Some(current_time));
+                    current_time
+
+                });
+                let relative_start_ms = data.relative_start_ms();
+                let relative_time_ms  = current_time - absolute_start_ms + relative_start_ms;
+                data.on_animation_frame(relative_time_ms);
+            }
```

Comment:
I'd really like to have this block in a separate function. It's not intuitive to search for "code run for each frame" in constructor.

So, the `if lest Some(data)` should remain here, but the main logic should be a separate function - maybe as member of ContinuousAnimatorData?

Reply:
It makes sense. I moved the whole block to `ContinuousAnimatorData`'s `on_animation_frame`.